todo_app/data/trello_items.py

```diff
@@ -0,0 +1,146 @@
+import os
+from todo_app.data.item import Item, ItemStatus
+from typing import List, Optional
+
+import requests
+
+query_params = {
+    'token': os.environ['TRELLO_API_TOKEN'],
+    'key': os.environ['TRELLO_API_KEY']
+}
+
+base_uri = "https://api.trello.com/1"
+
+def get_todo_board_id() -> str:
+    configured_todo_board_name = os.environ['TRELLO_API_TODO_BOARD_NAME']
+    response = requests.get(f'{base_uri}/members/me/boards', params=query_params).json()
+    boards_filtered_by_name = [x for x in response if x['name'] == configured_todo_board_name]
+    todo_board = next(iter(boards_filtered_by_name), None)
+
+    if (todo_board is None):
+        raise ValueError(f'Could not find configured to do board: {configured_todo_board_name}')
+
+    return todo_board['id']
+
+def get_lists() -> List[dict]:
+    todo_board_id = get_todo_board_id()
+    return requests.get(f'{base_uri}/boards/{todo_board_id}/lists', params=query_params).json()
+
+def get_list_id_or_throw(lists: List[dict], list_name: str) -> str:
+    list = next(iter([x for x in lists if x['name'] == list_name]), None)
+    if (list is None):
+        raise ValueError(f'Could not find list "{list_name}". Make sure you have this configured on your trello board')
+    return list['id']
+
+def get_items_from_list(list_id: str, status: str) -> List[Item]:
+    cards = requests.get(f'{base_uri}/lists/{list_id}/cards', params=query_params).json()
+    return [Item(id=card['id'], title=card['name'], status=status) for card in cards]
+
+def create_todo_item(title: str):
+    lists = get_lists()
+    to_do_list_id = get_list_id_or_throw(lists, "To Do")
+    response = requests.post(f'{base_uri}/cards', params={**query_params, 'name': title, 'idList': to_do_list_id})
+
+    if response.status_code != 200:
+        raise ValueError(f'Failed to create todo item for "{title}". Received status code {response.status_code}')
```

Comment:
Nice error checking here, worth noting that requests does have an appropriate built in alternative if that's helpful - `response.raise_for_status()`
That has some small advantages - like us not needing to worry about explicitly allowing for Trello to return a 2XX (but not 200) success code instead